### DISCUSSION TITLE
Can't install waydroid on steamdeck - error I don't understand

### DISCUSSION THREAD

**Discussion by mangonebula** [waydroid/waydroid#1375] [Q&A]
I tried to install waydroid following the instructions on this website and it failed. I didn't think to save the konsole text to explore the error.
I tried to uninstall it, but I get a message that the uninstall.sh file doesn't exist.
I can't open the waydroid_toolbox file on the desktop. I just get a window that asks me to select the program to open the file in.
I tried to just do a new install, but I get the error that I don't understand. I included a file that has a copy of the konsole.
When I try to do the git clone command, it says it can't copy the files because there's already something there.

I'm at a loss, because this is the extent of my skills with this kind of stuff.

Can you help me clean up waydroid and get it installed properly?

SteamOS Waydroid Installer Script by ryanrudolf
https://github.com/ryanrudolfoba/SteamOS-Waydroid-Installer
Script is running in Desktop Mode.
Checking if kernel is supported.
6.1.52-valve16-1-neptune-61 is supported. Proceed to next step.
Please enter current sudo password:
Checking if the sudo password is correct.
Sudo password is good!
Cloning into '/home/deck/AUR/waydroid/waydroid_script'...
remote: Enumerating objects: 717, done.
remote: Counting objects: 100% (327/327), done.
remote: Compressing objects: 100% (129/129), done.
remote: Total 717 (delta 260), reused 237 (delta 198), pack-reused 390
Receiving objects: 100% (717/717), 21.74 MiB | 17.21 MiB/s, done.
Resolving deltas: 100% (438/438), done.
Casualsnek repo has been successfully cloned!
Warning: The rootfs is already read-write!
Nothing is performed.
==> Appending keys from archlinux.gpg...
==> Appending keys from holo.gpg...
==> Updating trust database...
gpg: next trustdb check due at 2024-06-29
pacman keyring has been initialized!
binder kernel module has been installed!
loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (7) dnsmasq-2.89-1 libgbinder-1.1.35-1 libglibutil-1.0.74-1 lxc-1:5.0.2-1 python-gbinder-1.1.2-1 waydroid-1.4.2-1
wlroots-0.16.2-1

Total Installed Size: 31.91 MiB

:: Proceed with installation? [Y/n]
(7/7) checking keys in keyring [################################################] 100%
(7/7) checking package integrity [################################################] 100%
(7/7) loading package files [################################################] 100%
(7/7) checking for file conflicts [################################################] 100%
(7/7) checking available disk space [################################################] 100%
:: Processing package changes...
(1/7) installing wlroots [################################################] 100%
Optional dependencies for wlroots
xorg-xwayland: Xwayland support [installed]
(2/7) installing dnsmasq [################################################] 100%
(3/7) installing lxc [################################################] 100%
Optional dependencies for lxc
dnsmasq: lxc-net.service [installed]
lua [installed]
lua-filesystem: lxc-top
lua-alt-getopt: lxc-top
(4/7) installing libglibutil [################################################] 100%
(5/7) installing libgbinder [################################################] 100%
(6/7) installing python-gbinder [################################################] 100%
(7/7) installing waydroid [################################################] 100%
Optional dependencies for waydroid
python-pyclip: share clipboard with container
:: Running post-transaction hooks...
(1/7) Creating system user accounts...
(2/7) Reloading system manager configuration...
(3/7) Creating temporary files...
/usr/lib/tmpfiles.d/steamos.conf:6: Duplicate line for path "/root", ignoring.
/usr/lib/tmpfiles.d/steamos.conf:23: Duplicate line for path "/var/empty", ignoring.
/usr/lib/tmpfiles.d/tmp.conf:12: Duplicate line for path "/var/tmp", ignoring.
/usr/lib/tmpfiles.d/var.conf:19: Duplicate line for path "/var/cache", ignoring.
(4/7) Arming ConditionNeedsUpdate...
(5/7) Reloading system bus configuration...
(6/7) Updating icon theme caches...
(7/7) Updating the desktop file MIME type cache...
waydroid and cage has been installed!
ln: failed to create symbolic link '/home/deck/Desktop/Waydroid-Toolbox': File exists
Config file missing. Lets configure waydroid.
Traceback (most recent call last):
File "/usr/bin/waydroid", line 6, in
import tools
File "/usr/lib/waydroid/tools/init.py", line 8, in
import dbus.mainloop.glib
ModuleNotFoundError: No module named 'dbus'
Waydroid did not initialize correctly. Performing cleanup!
checking dependencies...
:: networkmanager optionally requires dnsmasq: connection sharing

Packages (7) dnsmasq-2.89-1 libgbinder-1.1.35-1 libglibutil-1.0.74-1 lxc-1:5.0.2-1 python-gbinder-1.1.2-1 waydroid-1.4.2-1
wlroots-0.16.2-1

Total Removed Size: 31.91 MiB

:: Do you want to remove these packages? [Y/n]
:: Processing package changes...
(1/7) removing wlroots [################################################] 100%
(2/7) removing waydroid [################################################] 100%
(3/7) removing lxc [################################################] 100%
(4/7) removing dnsmasq [################################################] 100%
(5/7) removing python-gbinder [################################################] 100%
(6/7) removing libgbinder [################################################] 100%
(7/7) removing libglibutil [################################################] 100%
:: Running post-transaction hooks...
(1/5) Reloading system manager configuration...
(2/5) Arming ConditionNeedsUpdate...
(3/5) Reloading system bus configuration...
(4/5) Updating icon theme caches...
(5/5) Updating the desktop file MIME type cache...
Cleanup completed! Try running the install script again! Goodbye!
(deck@steamdeck steamos-waydroid-installer)$

Waydroid version
latest

Device
Linux Desktop

Operating System
Steam OS Holo 3.5.17

Kernel version
6.1.52-valve16-1-neptune-61

Desktop Environment
kde plasma

GPU
Van Gogh 100-000000405

**Comment by LukasGossmann** [comment]
In the log file it says this:

```
File "/usr/bin/waydroid", line 6, in
import tools
File "/usr/lib/waydroid/tools/init.py", line 8, in
import dbus.mainloop.glib
ModuleNotFoundError: No module named 'dbus'
```
Notice the `ModuleNotFoundError`.
To me it looks like you are probably missing this package: [dbus-python](https://archlinux.org/packages/extra/x86_64/dbus-python/)
Probably you'll need to download the file and install it manually beforehand.

I can't really tell you how though as I don't use arch linux so i don't really know my way around their package manager.
Also please be careful with installing packages manually / from the repo.
Previously i tried installing some packages that also include the header files to compile some code on the SteamDeck and ended up damaging the install (unable to connect to wifi, unstable switching between desktop and game mode).

If that happens to you here is a guide how to restore the steamdeck os: [Steam Deck Recovery Instructions](https://help.steampowered.com/en/faqs/view/1B71-EDF2-EB6D-2BB3)
Make sure to back up your data beforehand just in case. Also downloaded games were deleted in my case (not sure why).

Oh and probably questions about the install script are better asked on the [SteamOS-Waydroid-Installer](https://github.com/ryanrudolfoba/SteamOS-Waydroid-Installer) repo since that script is not a part of waydroid.

**Comment by 6666GGGGGG** [comment]
I had this error and with chatgpt I was able to fix it with these step by step instructions



Step 1: Manually remove any old Waydroid state

Stop Waydroid and its container service:

```bash
waydroid session stop 2>/dev/null
sudo systemctl stop waydroid-container.service 2>/dev/null
sudo systemctl disable waydroid-container.service 2>/dev/null
sudo systemctl reset-failed
```

Try removing the package:

```bash
sudo pacman -Rns waydroid
```

If you get:

```bash
error: failed to init transaction (unable to lock database)
error: could not lock database: Read-only file system
```

unlock SteamOS first:

```bash
sudo steamos-readonly disable
sudo pacman -Rns waydroid
```

If you get:

```bash
error: target not found: waydroid
```

that means the package is already gone. Continue with leftover cleanup.

Remove leftover Waydroid directories:

```bash
sudo rm -rf /var/lib/waydroid
sudo rm -rf /var/cache/waydroid
sudo rm -rf /usr/share/waydroid
rm -rf ~/.local/share/waydroid
rm -rf ~/.config/waydroid
```

If you also had a leftover folder like your repack folder:

```bash
sudo rm -rf "/home/deck/REPACKS/data (2)"
```

If that folder refuses to delete, run:

```bash
ls -ld "/home/deck/REPACKS/data (2)"
lsattr -d "/home/deck/REPACKS/data (2)" 2>/dev/null
sudo chown -R deck:deck "/home/deck/REPACKS/data (2)"
sudo chattr -i "/home/deck/REPACKS/data (2)" 2>/dev/null
sudo rm -rf "/home/deck/REPACKS/data (2)"
```

Step 2: Confirm Waydroid is actually removed

Run:

```bash
which waydroid
pacman -Qs waydroid
ls /var/lib | grep waydroid
```

Expected good state:

* `which waydroid` returns nothing
* `pacman -Qs waydroid` returns nothing
* `ls /var/lib | grep waydroid` returns nothing

Step 3: Reinstall using a fresh copy of the installer

Use a new clone of the installer:

```bash
cd ~
rm -rf SteamOS-Waydroid-Installer
git clone https://github.com/ryanrudolfoba/SteamOS-Waydroid-Installer.git
cd SteamOS-Waydroid-Installer
chmod +x steamos-waydroid-installer.sh
./steamos-waydroid-installer.sh
```

Step 4: If the installer fails at “Building and installing binder module from source”

This was your main failure.

The symptom in your log was:

* running kernel: `6.11.11.valve26-1`
* headers installed: `6.11.11.valve27-1`

Check that with:

```bash
uname -r
pacman -Q linux-neptune-611 linux-neptune-611-headers
ls /lib/modules
```

Your mismatch looked like this:

* `uname -r` → valve26
* `linux-neptune-611` → valve26
* `linux-neptune-611-headers` → valve27

That breaks DKMS, so binder does not build.

Step 5: Fix the kernel/header mismatch

First try the simplest path:

```bash
sudo steamos-readonly disable
sudo pacman -Syu
sudo reboot
```

After reboot, verify both kernel and headers match:

```bash
uname -r
pacman -Q linux-neptune-611 linux-neptune-611-headers
```

You want both package versions aligned on the same `valveXX` release.

In your case, after reboot you moved onto valve27, which was correct.

Step 6: Rebuild binder manually if the installer still fails

Remove the broken DKMS package:

```bash
sudo pacman -Rns binder_linux-dkms
```

Then rebuild it cleanly:

```bash
sudo steamos-readonly disable
sudo pacman -S --needed linux-neptune-611 linux-neptune-611-headers dkms base-devel fakeroot debugedit plymouth
cd /tmp
rm -rf binder_linux-dkms
git clone https://aur.archlinux.org/binder_linux-dkms.git
cd binder_linux-dkms
makepkg -f
sudo pacman -U --noconfirm ./binder_linux-dkms-*.pkg.tar.zst
```

Force DKMS to build for the currently running kernel:

```bash
sudo dkms autoinstall -k "$(uname -r)"
sudo depmod -a
sudo modprobe binder_linux device=binder,hwbinder,vndbinder
lsmod | grep binder
```

The successful output you got was:

```bash
binder_linux          262144  0
```

That line means the hard part is fixed.

Step 7: Rerun the Waydroid installer

Once binder is loading correctly, run the installer again from a fresh clone:

```bash
cd ~
rm -rf SteamOS-Waydroid-Installer
git clone https://github.com/ryanrudolfoba/SteamOS-Waydroid-Installer.git
cd SteamOS-Waydroid-Installer
chmod +x steamos-waydroid-installer.sh
./steamos-waydroid-installer.sh
```

At this point it should proceed past the binder stage and complete.

Step 8: Verify Waydroid works

Check status:

```bash
waydroid status
```

Start the session:

```bash
waydroid session start
```

Open the UI:

```bash
waydroid show-full-ui
```

Optional cleanup and protection

If you want to turn SteamOS readonly protection back on afterward:

```bash
sudo steamos-readonly enable
```

Short diagnosis of each issue you hit

Your uninstall issue:

* `pacman` failed because SteamOS root was readonly

Your package removal confusion:

* `target not found: waydroid` meant the package was already removed

Your reinstall failure:

* binder DKMS was building against `valve27` headers while the Deck was still running `valve26`

Your final fix:

* boot into the matching kernel version, rebuild `binder_linux-dkms`, confirm `binder_linux` loads, then rerun the installer

**Reply by katsumakie8-create** [reply]
yoo thank you so much!!!!
